include/GoogleSync/GoogleSyncBase.php

```diff
@@ -0,0 +1,1049 @@
+        }
+
+        // We do it this way so we also get deleted meetings
+        $query = "SELECT id FROM meetings WHERE assigned_user_id = '" . $userId . "' AND date_start <= now() + interval 3 month";
+        $result = $this->db->query($query);
+
+        $meetings = array();
+        while ($row = $this->db->fetchByAssoc($result)) {
+            $meeting = BeanFactory::getBean('Meetings');
+            if (!$meeting) {
+                throw new GoogleSyncException('Unable to get Meetings bean.', GoogleSyncException::UNABLE_TO_RETRIEVE_MEETING);
+            }
+            $meeting->retrieve($row['id'], true, false);
+            $meetings[] = $meeting;
+        }
+        return $meetings;
+    }
+
+    /**
+     * Set user's google calendar id
+     *
+     *
+     * @return bool|int Success/Failure calendar ID if success
+     */
+    protected function setUsersGoogleCalendar()
+    {
+
+        // Make sure we have a Google Calendar Service instance
+        if (!$this->isServiceExists()) {
+            return false;
+        }
+
+        // get list of users calendars
+        $calendarList = $this->gService->calendarList->listCalendarList();
+
+        // find the id of the 'SuiteCRM' calendar ... in the future, this will set the calendar of the users choosing.
+        $this->calendarId = $this->getSuiteCRMCalendar($calendarList);
+
+        // if the SuiteCRM calendar doesn't exist... Create it!
+        if (!$this->isCalendarExists()) {
+            // TODO: It should be an exception?
+            $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to find the SuiteCRM Google Calendar, Creating it!');
+            $calendar = new Google_Service_Calendar_Calendar();
+            $calendar->setSummary('SuiteCRM');
+            $calendar->setTimeZone($this->timezone);
+
+            $createdCalendar = $this->gService->calendars->insert($calendar);
+            $this->calendarId = $createdCalendar->getId();
+        }
+
+        // Final check to make sure we have an ID
+        if (!$this->isCalendarExists()) {
+            return false;
+        }
+        return $this->calendarId;
+    }
+
+    /**
+     * find the id of the 'SuiteCRM' calendar ... in the future, this will return the calendar of the users choosing.
+     *
+     * @param Google_Service_Calendar_CalendarList $calendarList
+     * 
+     * @return string|null Matching Google Calendar ID or null.
+     */
+    protected function getSuiteCRMCalendar(Google_Service_Calendar_CalendarList $calendarList)
+    {
+        foreach ($calendarList->getItems() as $calendarListEntry) {
+            if ($calendarListEntry->getSummary() == 'SuiteCRM') {
+                $return = $calendarListEntry->getId();
+                break;
+            }
+        }
+        if (empty($return)) { // TODO: this empty-check is not neccessary, return the getId() results as it is.
```

Comment:
@gymad
I'll have to refresh my memory on this. It may be that getId() returns an empty string or array, and I need a null.